src/stores/obsPortalData.js

```diff
@@ -82,6 +82,18 @@ export const useObsPortalDataStore = defineStore('obsPortalData', {
         }
       })
     },
+    async fetchSelectedObservationDetails (requestId) {
+      const configurationStore = useConfigurationStore()
+      const userDataStore = useUserDataStore()
+      const username = userDataStore.username
+      await fetchApiCall({
+        url: configurationStore.observationPortalUrl + `requestgroups/?state=PENDING&user=${username}&request_id=${requestId}`,
```

Comment:
It looks like you just want to get a specific request, not the whole request group? I assume you are using the requestgroups API since the requests API does not have the filters you want on state and user? If that is the case, we can add the filters to support that into the Observation Portal.

Reply:
Okay I'll look into doing this next time im in the codebase